### PR TITLE
Debian mate

### PR DESCRIPTION
**Pull request by jvonau**
### Fixes bug:
br0 handling in #2658 
### Description of changes proposed in this pull request:
enable systemd-networkd and install systemd-dispatcher for Debian 11 when NetworkManager is installed usually with a desktop to support br0, similar to Ubuntu desktops' layout that work fine just expanded coverage now.

### CODE REVIEW

**Comment by holta** [comment]
What HW/OS/networking layouts are most useful for testing this, if others can help?

**Comment by holta** [comment]
Is this PR possibly also helpful for Ubuntu 21.04?

(Ubuntu "Hirsute Hippo" 21.04 is currently in Beta, with its final release expected 2 weeks from today on Earth Day 2021-04-22.)

**Comment by jvonau** [comment]
> Is this PR possibly also helpful for Ubuntu 21.04?
> 
> (Ubuntu "Hirsute Hippo" 21.04 is currently in Beta, with its final release expected 2 weeks from today on Earth Day 2021-04-22.)

No, it's a specific fix for Mate Desktop (NetworkManager backend in use) on Debian in the above noted issue.

**Comment by holta** [comment]
Let's not lose valuable work like this PR.

(Which ideally should be re-opened, if others are Ok with that?)

**Comment by jvonau** [comment]
> Let's not lose valuable work like this PR.

Why just this PR? were not all my contributions valuable to the project?

> (Which ideally should be re-opened, if others are Ok with that?)

Do as you always do as "primary developer", whatever you please. Just remember to not alienate your future developers in the same way as I have been.

**Comment by holta** [comment]
To become more sustainable for the decade to come, it's critical that IIAB support fewer OS's very well _(rather than many OS's very poorly)._  The result will be to help many more communities, and orders of magnitude more children.  (And avoid death-by-a-thousand-cuts!)

Perhaps this is blindingly obvious.  (Nevertheless it's important this eternal tension be clearly and universally stated, especially in places like GitHub, where the FOSS community sometimes favors tinkering-for-the-sake-of-tinkering &mdash; rather than needed consolidation/productization = the "give me something Dead Simple" goal that NGO's and most every educator are constantly asking for.)

My assumption is that Debian MATE and/or Ubuntu MATE could potentially be very high-impact in future (especially on Raspberry Pi?) if this momentum one day unites with something like Ubuntu, Debian or RaspiOS &mdash; possibly achieving _mass adoption_ similar to [Linux Mint](https://en.wikipedia.org/wiki/Linux_Mint)?  e.g. the possibility of genuine popularity among non-technical users &mdash; backstopped by very broad global community support on sites like StackOverflow, StackExchange, etc (when things inevitably go wrong, Murphy's Law of UX / interface design ;)

In short, most all IIAB implementers stick with 32-bit RaspiOS in 2021 &mdash; but let's open and take seriously this PR if it advances and supports community decision-making on these eternal but not-easy topics.

**Comment by holta** [comment]
> What HW/OS/networking layouts are most useful for testing this, if others can help?

**Comment by jvonau** [comment]
> To become more sustainable for the decade to come, it's critical that IIAB support fewer OS's very well _(rather than many OS's very poorly)._ The result will be to help many more communities, and orders of magnitude more children. (And avoid death-by-a-thousand-cuts!)
>

Why have a call for testing on an OS (Debian Desktop) that **you** don't plan on supporting?
  
> Perhaps this is blindingly obvious. (Nevertheless it's important this eternal tension be clearly and universally stated, especially in places like GitHub, where the FOSS community sometimes favors tinkering-for-the-sake-of-tinkering — rather than needed consolidation/productization = the "give me something Dead Simple" goal that NGO's and most every educator are constantly asking for.)
> 

Now my contributions have been reduced to mere tinkering? You have some nerve.
 
> My assumption is that Debian MATE and/or Ubuntu MATE could potentially be very high-impact in future (especially on Raspberry Pi?) if this momentum one day unites with something like Ubuntu, Debian or RaspiOS — possibly achieving _mass adoption_ similar to [Linux Mint](https://en.wikipedia.org/wiki/Linux_Mint)? e.g. the possibility of genuine popularity among non-technical users — backstopped by very broad global community support on sites like StackOverflow, StackExchange, etc (when things inevitably go wrong, Murphy's Law of UX / interface design ;)
> 

Until Mint produces an ARM version that is a pipe dream, and why is Mint supported? because it's based on Ubuntu. Why is Ubuntu supported? Because the netplan/init-cloud layout was figured out by guess who.

> In short, most all IIAB implementers stick with 32-bit RaspiOS in 2021 — but let's open and take seriously this PR if it advances and supports community decision-making on these eternal but not-easy topics.

Cool I'll just fork and concentrate on Ubuntu and fix any issue in my tree.

**Comment by holta** [comment]
> Now my contributions have been reduced to mere tinkering?

Definitely not.

I'm addressing this universal (and incredibly serious) risk in the ethos of almost all FOSS communities — that we all need to avoid — allowing for impact to be very real.

**Comment by holta** [comment]
If this fixes https://github.com/iiab/iiab/issues/2749#issuecomment-825650153 that is great news.

**Comment by jvonau** [comment]
Since this is aimed at Debian's desktop, the first course of action would be to test this on other platforms to ensure there is no harm to those variants. Now that no illness was detected in those variants, merge into master, wait for somebody with Debian desktop with 2 network interfaces to install, should the install blow up, needs more work, silence for feedback it works fine.

**Comment by jvonau** [comment]
Now about some transparency when it comes to the "**NGOs and most every educator**" that you keep mentioning, but never disclose.

**Comment by holta** [comment]
@deldesir here's another possible IIAB install test if you have time:

```
curl d.iiab.io/install.txt | sudo bash -s 2693
```

But possibly testing on 64-bit RaspiOS with Desktop is not enough?  (An OS that advertises itself as Debian FWIW...)

As it's likely nobody has immediate access to a Desktop version of Debian 11 (with 2 network interfaces) as outlined [above](https://github.com/iiab/iiab/pull/2693#issuecomment-825703850).

**Comment by jvonau** [comment]
> But possibly testing on 64-bit RaspiOS with Desktop is not enough? (An OS that advertises itself as Debian FWIW...)
>

Depends on what local_facts.fact detects OS as, if it was built with pi-gen there should be the same marker file present, rpi_model isn't tied to any os. Testing this PR on any RaspiOS or Ubuntu would be testing "no harm here"
 
> As it's likely nobody has immediate access to a Desktop version of Debian 11 (with 2 network interfaces) as outlined [above]

Any laptop should work given drivers that work and support is in the firmware, the fix is just not for Debian 11 the request for testing in 2658 was for 10.7. At any rate the risk of breaking anything is very slim, Ubuntu's Desktops' are already being handled in the exactly the same manor very successfully. Too bad the end users are asked to test, and any fix is left to rot, just a waste of my time.

**Comment by holta** [comment]
Would it be helpful to test this PR on the new 2021-04-09 64-bit RaspiOS with Desktop on RPi 4?

**Comment by holta** [comment]
> Would it be helpful to test this PR on the new 2021-04-09 64-bit RaspiOS with Desktop on RPi 4?

Either way, it sounds like it can't hurt and might possibly shed some light here, so I kicked this off:

```
curl d.iiab.io/install.txt | sudo bash -s 2693
```

Install should take 1-2 hours to complete here: 10.8.0.50 = 184-rpi4-64bitRaspDesk-BIG-2693

**Comment by holta** [comment]
>> But possibly testing on 64-bit RaspiOS with Desktop is not enough? (An OS that advertises itself as Debian FWIW...)

> Depends on what local_facts.fact detects OS as, if it was built with pi-gen there should be the same marker file present, rpi_model isn't tied to any os. Testing this PR on any RaspiOS or Ubuntu would be testing "no harm here"

Install hasn't progressed enough to make iiab-diagnostics meaningful, but 10.8.0.50 (2021-04-09 64-bit RaspiOS with Desktop) shows these basic details:

```
root@box:~# uname -a
Linux box.lan 5.10.17-v8+ #1403 SMP PREEMPT Mon Feb 22 11:37:54 GMT 2021 aarch64 GNU/Linux

root@box:~# more /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

root@box:~# more /etc/debian_version
10.9

root@box:/opt/iiab/iiab# scripts/local_facts.fact
cat: /proc/device-tree/mfg-data/MN: No such file or directory
scripts/local_facts.fact: line 88: warning: command substitution: ignored null byte in input
Failed to get unit file state for NetworkManager.service: No such file or directory
{"stage"                  : "1",
"dhcpcd"                  : "enabled",
"network_manager"         : "none",
"systemd_networkd"        : "disabled",
"iiab_branch"             : "master",
"iiab_commit"             : "d44da126736def47467e77ca9f112fd8d4b9a950",
"xo_model"                : "none",
"rpi_model"               : "Raspberry Pi 4 Model B Rev 1.2",
"ansible_version"         : "2.10.8",
"os"                      : "raspbian",
"os_ver"                  : "raspbian-10"}

**Comment by holta** [comment]
BIG-sized IIAB install completed on RPi 4 at 10.8.0.50

It needs more spot checking of apps, but so far all looks good, including basic testing of its internal WiFi hotspot: https://github.com/iiab/iiab/issues/2748#issuecomment-828855063

Finally its [iiab-diagnostics](https://github.com/iiab/iiab/blob/master/scripts/iiab-diagnostics.README.md) summary details are uploaded here: http://sprunge.us/horjUV?en

**Comment by jvonau** [comment]
Looks a little funky to me but that would have nothing to do with this PR. You sure the wifi is stable? 
>2446 Apr 28 23:24:43 box hostapd: ap0: STA 8a:d4:52:48:3a:94 IEEE 802.11: associated
2447 Apr 28 23:24:43 box hostapd: ap0: STA 8a:d4:52:48:3a:94 RADIUS: starting accounting session 2D1CF3DC77777999
2448 Apr 28 23:24:46 box dnsmasq-dhcp[1133]: DHCPDISCOVER(br0) 8a:d4:52:48:3a:94 
2449 Apr 28 23:24:46 box dnsmasq-dhcp[1133]: DHCPOFFER(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2450 Apr 28 23:24:46 box dnsmasq-dhcp[1133]: DHCPDISCOVER(br0) 8a:d4:52:48:3a:94 
2451 Apr 28 23:24:46 box dnsmasq-dhcp[1133]: DHCPOFFER(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2452 Apr 28 23:24:46 box dnsmasq-dhcp[1133]: DHCPDISCOVER(br0) 8a:d4:52:48:3a:94 
2453 Apr 28 23:24:46 box dnsmasq-dhcp[1133]: DHCPOFFER(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2454 Apr 28 23:24:46 box dnsmasq-dhcp[1133]: DHCPREQUEST(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2455 Apr 28 23:24:46 box dnsmasq-dhcp[1133]: DHCPACK(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2456 Apr 28 23:24:57 box dnsmasq-dhcp[1133]: DHCPREQUEST(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2457 Apr 28 23:24:57 box dnsmasq-dhcp[1133]: DHCPACK(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2458 Apr 28 23:24:57 box dnsmasq-dhcp[1133]: DHCPREQUEST(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2459 Apr 28 23:24:57 box dnsmasq-dhcp[1133]: DHCPACK(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2460 Apr 28 23:24:57 box dnsmasq-dhcp[1133]: DHCPREQUEST(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2461 Apr 28 23:24:57 box dnsmasq-dhcp[1133]: DHCPACK(br0) 172.18.102.65 8a:d4:52:48:3a:94 
2462 Apr 28 23:24:58 box hostapd: ap0: STA 72:2d:4f:ff:4c:42 IEEE 802.11: disassociated
2463 Apr 28 23:25:38 box last message repeated 2 times
2464 Apr 28 23:26:38 box last message repeated 3 times
2465 Apr 28 23:27:38 box last message repeated 3 times
2466 Apr 28 23:28:38 box last message repeated 3 times
2467 Apr 28 23:29:38 box last message repeated 3 times
2468 Apr 28 23:30:38 box last message repeated 3 times
2469 Apr 28 23:31:38 box last message repeated 3 times
2470 Apr 28 23:32:39 box last message repeated 3 times
2471 Apr 28 23:33:39 box last message repeated 3 times
2472 Apr 28 23:34:39 box last message repeated 3 times
2473 Apr 28 23:35:39 box last message repeated 3 times
2474 Apr 28 23:36:39 box last message repeated 3 times
2475 Apr 28 23:37:39 box last message repeated 3 times
2476 Apr 28 23:38:39 box last message repeated 3 times
2477 Apr 28 23:39:39 box last message repeated 3 times
2478 Apr 28 23:40:39 box last message repeated 3 times
2479 Apr 28 23:41:39 box last message repeated 3 times
2480 Apr 28 23:42:39 box last message repeated 3 times
2481 Apr 28 23:43:39 box last message repeated 3 times
2482 Apr 28 23:44:39 box last message repeated 3 times
2483 Apr 28 23:45:39 box last message repeated 3 times
2484 Apr 28 23:46:39 box last message repeated 3 times
2485 Apr 28 23:47:39 box last message repeated 3 times
2486 Apr 28 23:48:39 box last message repeated 3 times

**Comment by jvonau** [comment]
> BIG-sized IIAB install completed on RPi 4 at 10.8.0.50
> 
> It needs more spot checking of apps, but so far all looks good, including basic testing of its internal WiFi hotspot: [#2748 (comment)](https://github.com/iiab/iiab/issues/2748#issuecomment-828855063)

Other apps really don't come into play when evaluating this PR, this is strictly a 'does the network function' question. Quick testing on Ubuntu as the other supported distro and this should be good to go.

**Comment by holta** [comment]
> Looks a little funky to me but that would have nothing to do with this PR. You sure the wifi is stable?

Over about 7-8 hours of uptime, WiFi has appeared completely stable and reliable.

Caveat: I had to click twice on the Android client machine, the very first time it was associating with SSID 184-rpi4-64bitRaspDesk-BIG-2693

(But there hasn't been a single hiccup since then, that I know of.  I re-associated about 3 times, and clicked around http://box.lan over a few minutes each time.  It all seems to work great!)

**Comment by holta** [comment]
> this is strictly a 'does the network function' question. Quick testing on Ubuntu as the other supported distro and this should be good to go.

This PR is now being installed (with 4 others) on Ubuntu Server 21.04 on RPi 4 on 10.8.0.22 = 182-rpi4-u21-srv-BIG-rollup using:

```
curl d.iiab.io/install.txt | sudo bash -s 2693 2743 2759 2760 2761
```

Ref PR's: #2743, #2759, #2760, #2761

**Comment by jvonau** [comment]
Left a note in 2759 that was missed.

**Comment by holta** [comment]
> Left a note in 2759 that was missed.

Does the output of [iiab-diagnostics](https://github.com/iiab/iiab/blob/master/scripts/iiab-diagnostics.README.md) need immediate improvements possibly?

**Comment by holta** [comment]
> This PR is now being installed (with 4 others) on Ubuntu Server 21.04 on RPi 4 on 10.8.0.22 = 182-rpi4-u21-srv-BIG-rollup using:
> 
> ```
> curl d.iiab.io/install.txt | sudo bash -s 2693 2743 2759 2760 2761
> ```

Install completed & rebooted, with [iiab-diagnotics](https://github.com/iiab/iiab/blob/master/scripts/iiab-diagnostics.README.md) here: http://sprunge.us/WTm308?en
